### ISSUE TITLE
Add the workspace name to the title of the workspace chats

### ISSUE THREAD

**Issue by davidcardoza** [Expensify/App#49624]
### Background

We’ve been discussing making the workspace name appear first in all workspace rooms to improve efficiency when identifying which company we are communicating with. The current process requires multiple clicks (#admins > #admins details > members) to see the domain of a new lead, which is inefficient. More background can be found in this Slack conversation - https://expensify.slack.com/archives/C07HPDRELLD/p1726677008605379

**Design**
The plan is to adda workspace name to the front of the workspca rooms, for example, "My Workspace • #admins". This will allow the workspace name to be visible without needing to click through multiple steps. 

**Example:**
![image](https://github.com/user-attachments/assets/3367d155-d6c5-4236-9dfa-5da01530cd78)

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?

New feature request

### What changes do you think we should make in order to solve the problem?


1. We need to add a new condition [here](https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/libs/ReportUtils.ts#L3721)  to add the workspace name in the title base on `chatType`.

```
    if (isUserCreatedPolicyRoom(report) || isDefaultRoom(report)) {
        formattedName = `${report?.policyName ?? ''} • ${report?.reportName ?? ''}`;
    }
```

https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/libs/ReportUtils.ts#L3721

2. As I see in the design, we will not show the subtitle as the workspace name so we need to add early return empty string [here](https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/libs/ReportUtils.ts#L3860)

```
    if (isChatThread(report) || isUserCreatedPolicyRoom(report) || isDefaultRoom(report)) {
        return '';
    }
```

https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/libs/ReportUtils.ts#L3860

3. We also don't need the additional text in `HeaderView` [here](https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/pages/home/HeaderView.tsx#L112)

```
    if (shouldShowSubtitle() || isPersonalExpenseChat || !policyName || !isEmptyObject(parentNavigationSubtitleData) || isSelfDM || ReportUtils.isUserCreatedPolicyRoom(report) || ReportUtils.isDefaultRoom(report)) {
            return null;
    }
```

https://github.com/Expensify/App/blob/ae5763960d53fbac8a95da0619e2c98c3ba9237f/src/pages/home/HeaderView.tsx#L112

### What alternative solutions did you explore? (Optional)

**Comment by mollfpr** [comment]
@nkdengineer, your first step does not seem to be working well. It got overridden from the next condition. https://github.com/Expensify/App/blob/05ca35c015b268e495fe94eff36c9eda7afb78bc/src/libs/ReportUtils.ts#L3850-L3852

---

@nyomanjyotisa Almost looks good. There are a couple of things that we need to work on:
1. There's still additional text in the header view
2. Looking at the design, there's no workspace name in the welcome text
![Screenshot 2024-09-27 at 19 15 28](https://github.com/user-attachments/assets/8b8583ba-008e-4dd1-94e6-7fed022277dd)
3. The workspace name is `undefined` when creating a room offline
![Screenshot 2024-09-27 at 19 13 57](https://github.com/user-attachments/assets/9a2769fa-108e-4dcb-ae8b-47248df20c3f)

---

I think we can go with @nyomanjyotisa [proposal](https://github.com/Expensify/App/issues/49624#issuecomment-2369976495) and fix the above issue while working on the PR.

🎀 👀 🎀 C+ reviewed!

**Comment by nkdengineer** [comment]
> @mollfpr your first step does not seem to be working well. It got overridden from the next condition.

@mollfpr I see @nyomanjyotisa's proposal also uses the same condition as mine. Why did you only inform me? 
And nyomanjyotisa's proposal does not cover all cases

**Comment by mollfpr** [comment]
>I see @nyomanjyotisa's proposal also uses the same condition as mine. Why did you only inform me?

Their proposal latest update proposal is come first from your proposal. Yours 03:11 UTC and theirs 02:22 UTC, I don't see anything wrong with that.


> And nyomanjyotisa's proposal does not cover all cases

![Screenshot 2024-09-30 at 11 24 47](https://github.com/user-attachments/assets/69f6c952-b64d-40c5-a5f3-c4ab50594462)

You're correct, but the result from your proposal wasn't close to the design either. So I don't have anything to asses.

**Comment by dubielzyk-expensify** [comment]
Default workspace names or long workspace names will clip the room name on mobile phones:

![CleanShot 2024-10-01 at 20 13 04@2x](https://github.com/user-attachments/assets/c4ff9b1a-f2be-491f-b7e6-dd2f10b96fc6)

cc @dubielzyk-expensify

**Comment by dannymcclain** [comment]
>Default workspace names or long workspace names will clip the room name on mobile phones

I brought this up more than once and advocated for putting the room name first, but it didn't seem like anyone outside the design team was concerned about this inevitability... Definitely feel free to bring it up again in Slack! Or we can let this play out and let people experience the issue for themselves.

**Comment by dubielzyk-expensify** [comment]
Yeah, I did the same. Let's push through then and see if this becomes a more widespread problem or not 👍

**Comment by davidcardoza** [comment]
Side note: I’d love to see the room name as a header in the LHN, with the chat preview as a subheader. I feel like this would keep the LHN way more organized and give more space for the chat preview, especially when the room name is too long.

Example:

>Workspace123 - #admins
What up, this my admins chat

>Workspace456 - #admins
Do you like peanut butter on your expenses?

I think a similar idea was discussed before in a thread somewhere, but nothing came of it.

**Comment by trjExpensify** [comment]
> Default workspace names or long workspace names will clip the room name on mobile phones:
> 
> 
> 
> ![CleanShot 2024-10-01 at 20 13 04@2x](https://github.com/user-attachments/assets/c4ff9b1a-f2be-491f-b7e6-dd2f10b96fc6)
> 
> 
> 
> cc @Expensify/design 

Dang, that's really not great IMO. To further illustrate the point, add a bunch of messages in those chats and you won't be able to see the name at all anywhere. Super disorientating.

**Comment by JmillsExpensify** [comment]
Woof, I'm in the same spot. I think this creates more problems than it solves.

**Comment by mountiny** [comment]
I agree that we should ensure the room name is visible at the header at all times.

**Comment by trjExpensify** [comment]
Yeah, here's an example using our real UK expense policy. You'd have no clue what chat this is:

<img width="304" alt="image" src="https://github.com/user-attachments/assets/d7527263-7a18-4580-8aba-d4ef65e03c98">

**Comment by trjExpensify** [comment]
This issue OP doesn't seem to capture the full proposal discussed in the threads either: 

1. Don't include the workspace name when the user only has one workspace
2. Don't include the workspace name when filtered to a specific workspace via the workspace switcher

But even then, it breaks down for a policyExpenseChat title as the title would then become "Member name" which is exactly the same as the DM name for that member, so you can't distinguish between the two different chats by name.

If someone is motivated to advocate for this kind of change, it should go through the design doc process such that every case/report/chatType/flow that uses this logic is written down, mocked up and thought through. 

The logic for "chat headers", "avatars", "LHN rows", "participant selector" and such has historically been a nightmare to keep free of regressions, precisely because we keep making piecemeal "improvements" like this without taking a holistic approach to changes to them across the board. 

Some additional questions that come to mind:

- What happens when you have a room description? Are we dropping the "in <workspaceName>" which gets added to the first line of the chat header today? 
- How does it look in "Find" the chat switcher? Are we replacing the workspaceName in the second line with the lastMessage preview for all chatTypes now there? 
- How do the chats now render in the OptionSelector used in flows like `Submit expense` etc?

**Comment by dannymcclain** [comment]
>Side note: I’d love to see the room name as a header in the LHN, with the chat preview as a subheader. I feel like this would keep the LHN way more organized and give more space for the chat preview, especially when the room name is too long.

@davidcardoza I'm confused, isn't this how the LHN currently works? Or are you saying that you want focus mode to ALSO show this style the same way most recent mode does?

>If someone is motivated to advocate for this kind of change, it should go through the design doc process such that every case/report/chatType/flow that uses this logic is written down, mocked up and thought through.

@trjExpensify Please bring this back up in Slack if you haven't already done so. You're points are 100% valid and I agree with your comment about piecemeal improvements.

**Comment by JmillsExpensify** [comment]
Maybe we should create a new discussion thread and also revert this change until we have a holistic solution?

**Comment by mountiny** [comment]
We ended up reverting this PR, but I think its fair we will pay the contributors on this one as it was product decision down the line

**Comment by mountiny** [comment]
$250 to @mollfpr and @nyomanjyotisa

**Comment by puneetlath** [comment]
@nyomanjyotisa can you link me your upwork profile?

**Comment by puneetlath** [comment]
@nyomanjyotisa offer is here. Please ping me on this issue when you've accepted. 

https://www.upwork.com/nx/wm/offer/104447593

**Comment by puneetlath** [comment]
@nyomanjyotisa has been paid. @mollfpr please request on NewDot. Thanks y'all!

**Comment by garrettmknight** [comment]
$250 approved for @mollfpr
